pkg/build/api/v1/types.go

```diff
@@ -477,8 +478,23 @@ type SourceBuildStrategy struct {
 	// incremental flag forces the Source build to do incremental builds if true.
 	Incremental bool `json:"incremental,omitempty"`
 
-	// forcePull describes if the builder should pull the images from registry prior to building.
+	// forcePull describes if the builder should pull the images from registry
+	// prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// runtimeImage is an optional image that is used to run an application
+	// without unneeded dependencies installed.
+	RuntimeImage *kapi.ObjectReference `json:"runtimeImage,omitempty"`
+
+	// runtimeArtifacts holds the mapping between the artifacts paths in the
+	// builder image and the paths in the runtime image where they should be
+	// copied.
+	RuntimeArtifacts []ImageSourcePath `json:"runtimeArtifacts,omitempty"`
+
+	// runtimePullSecret is the name of a Secret that would be used for setting up
+	// the authentication for pulling the Docker runtime images from private
+	// Docker registries.
+	RuntimePullSecret *kapi.LocalObjectReference `json:"runtimePullSecret,omitempty"`
```

Comment:
see build/controller/strategy/util.go:setupDockerSecrets to see how user-provided secrets get setup as mount points and an env variable gets defined that points to the mount point, and then later we reference the env variable to find the mount point and locate the secret to do the actual pull.